modules/holidBidAdapter.js

```diff
@@ -0,0 +1,204 @@
+        },
+        site: {
+          page: getRefererInfo().page,
+          ref: getRefererInfo().ref,
+          domain: getRefererInfo().domain,
+        },
+        regs: getRegs(bidderRequest),
+      }
+
+      return {
+        method: 'POST',
+        url: ENDPOINT,
+        data: JSON.stringify(requestData),
+        bidId: bid.bidId,
+      }
+    })
+  },
+
+  interpretResponse: function (serverResponse, bidRequest) {
+    const bidResponses = []
+
+    if (!serverResponse.body.seatbid) {
+      return []
+    }
+
+    serverResponse.body.seatbid.map((response) => {
+      response.bid.map((bid) => {
+        const requestId = bidRequest.bidId
+        const auctionId = bidRequest.auctionId
+        const wurl = deepAccess(bid, 'ext.prebid.events.win')
+        const bidResponse = {
+          requestId,
+          cpm: bid.price,
+          width: bid.w,
+          height: bid.h,
+          ad: bid.adm,
+          creativeId: bid.crid,
+          currency: serverResponse.body.cur,
+          netRevenue: true,
+          ttl: TIME_TO_LIVE,
+        }
+
+        addWurl({ auctionId, requestId, wurl })
+
+        bidResponses.push(bidResponse)
+      })
+    })
+
+    return bidResponses
+  },
+
+  getUserSyncs(optionsType, serverResponse, gdprConsent, uspConsent) {
+    if (!serverResponse || serverResponse.length === 0) {
+      return []
+    }
+
+    const syncs = []
+
+    if (optionsType.iframeEnabled) {
+      const queryParams = []
+
+      queryParams.push('bidders=' + getBidders(serverResponse))
+      queryParams.push('gdpr=' + +gdprConsent.gdprApplies)
+      queryParams.push('gdpr_consent=' + gdprConsent.consentString)
+      queryParams.push('usp_consent=' + (uspConsent || ''))
+
+      let strQueryParams = queryParams.join('&')
+
+      if (strQueryParams.length > 0) {
+        strQueryParams = '?' + strQueryParams
+      }
+
+      syncs.push({
+        type: 'iframe',
+        url: COOKIE_SYNC_ENDPOINT + strQueryParams + '&type=iframe',
+      })
+
+      return syncs
+    }
+  },
+}
+
+function getImp(bid) {
+  const imp = {
+    ext: {
+      prebid: {
+        storedrequest: {
+          id: getBidIdParameter('adUnitID', bid.params),
+        },
+      },
+    },
+  }
+  const sizes =
+    bid.sizes && !Array.isArray(bid.sizes[0]) ? [bid.sizes] : bid.sizes
+
+  if (deepAccess(bid, 'mediaTypes.banner')) {
+    imp.banner = {
+      format: sizes.map((size) => {
+        return { w: size[0], h: size[1] }
+      }),
+    }
+  }
+
+  return imp
+}
+
+function getRegs(bidderRequest) {
+  const regs = {}
+
+  if (bidderRequest.gdprConsent) {
+    deepSetValue(
+      regs,
+      'ext.gdpr',
+      bidderRequest.gdprConsent.gdprApplies ? 1 : 0
+    )
+    deepSetValue(
+      regs,
+      'ext.gdprConsentString',
+      bidderRequest.gdprConsent.consentString || ''
+    )
+  }
+
+  if (bidderRequest.uspConsent) {
+    deepSetValue(regs, 'ext.us_privacy', bidderRequest.uspConsent)
+  }
+
+  return regs
```

Comment:
Why not just merge the whole regs object? Or the while ortb2 object?

Reply:
Great idea, thank you! I have updated the PR with merging the whole ortb2 object